docs/rules/0123/resource-definition-pattern.md

```diff
@@ -0,0 +1,88 @@
+---
+rule:
+  aip: 123
+  name: [core, '0123', resource-definition-pattern]
+  summary: Resource annotations should define a pattern.
+permalink: /123/resource-definition-pattern
+redirect_from:
+  - /0123/resource-definition-pattern
+---
+
+# Resource patterns
+
+This rule enforces that files that define a resource with the
+`google.api.resource_definition` annotation have a `pattern` defined, as
```

Comment:
I noticed that `google.api.resource_defintion` does not appear in [AIP-123](https://google.aip.dev/123). Should we create an issue for this @ [aip-dev/google.aip.dev](https://github.com/aip-dev/google.aip.dev)?

Reply:
I've opened https://github.com/aip-dev/google.aip.dev/issues/865